CONSTRUCT.md

```diff
@@ -318,31 +318,40 @@ Defaults to `${NAME} ${VERSION} (Python ${PYVERSION} ${ARCH})`.
 
 _required:_ no<br/>
 _type:_ string<br/>
-Path to a pre-install script. For Unix `.sh` installers, the shebang
-line is respected if present; otherwise, the script is run by the POSIX
-shell `sh`. Note that the use of a shebang can reduce the portability of
-the installer. Metadata about the installer can be found in the
-`${INSTALLER_NAME}`/`${INSTALLER_VER}`/`${INSTALLER_PLAT}` environment
-variables. This option is not supported for Windows `.exe` or macOS
-`.pkg` installers.
+Path to a pre-install script, run after the package cache has been set, but
+before the files are linked to their final locations. As a result, you should
+only rely on tools known to be available on most systems (e.g. `bash`, `cmd`,
+etc). See `post_install` for information about available environment variables.
+
+## `pre_install_desc`
+
+_required:_ no<br/>
+_type:_ string<br/>
+A description of the purpose of the supplied pre_install script. If this
+string is supplied and non-empty, then the Windows and macOS GUI installers
+will display it along with checkbox to enable or disable the execution of the
+script. If this string is not supplied, it is assumed that the script
```

Comment:
And what will happen on Linux, a Y/N prompt, or nothing?

Reply:
Nothing. These are meant to be used in the GUI installers, so they are not used at all in the shell installers. For the `sh` installers, users can disable scripts via the `--skip-scripts` command-line flag. Maybe we can clarify this in the docs. Thanks for the suggestion!